cookiecutter-django-ida/{{cookiecutter.repo_name}}/Dockerfile

```diff
@@ -42,29 +42,49 @@ RUN locale-gen en_US.UTF-8
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
-ENV DJANGO_SETTINGS_MODULE {{cookiecutter.project_name}}.settings.production
 
 EXPOSE {{cookiecutter.port}}
 RUN useradd -m --shell /bin/false app
 
 WORKDIR /edx/app/{{cookiecutter.repo_name}}
 
+RUN mkdir -p /edx/var/log
+
+# Code is owned by root so it cannot be modified by the application user.
+# So we copy it before changing users.
+USER app
+
+FROM app as dev
+ENV DJANGO_SETTINGS_MODULE {{cookiecutter.project_name}}.settings.devstack
+
 # Copy the requirements explicitly even though we copy everything below
 # this prevents the image cache from busting unless the dependencies have changed.
-COPY requirements/production.txt /edx/app/{{cookiecutter.repo_name}}/requirements/production.txt
+COPY requirements/development.txt /edx/app/{{cookiecutter.repo_name}}/requirements/development.txt
 
 # Dependencies are installed as root so they cannot be modified by the application user.
-RUN pip install -r requirements/production.txt
+RUN pip install -r requirements/development.txt
 
-RUN mkdir -p /edx/var/log
+# Gunicorn 19 does not log to stdout or stderr by default. Once we are past gunicorn 19, the logging to STDOUT need not be specified.
+CMD while true; do python ./manage.py runserver 0.0.0.0:8120; sleep 2; done
```

Comment:
Instead of hardcoding the port, use cookie-cutter input `{{cookiecutter.port}}`

Reply:
done

Reply:
> While enhancing native Dockerfiles to use to build docker images for devstack, we have set some standards which we want to replicate in cookiecutter here. From those standards was the use of `virtualenv` to install python requirements. We also need to set up `virtualenv` here so we have consistency among IDAs.

Thanks @iamsobanjaved for the suggestion. I've added virtual environment and tested its successful docker image build on local.